Source/VCGeneration/Checker.cs

```diff
@@ -299,13 +300,14 @@ private void WaitForOutput(object dummy)
     {
       lock (this)
       {
-        try
-        {
+        try {
           outcome = thmProver.CheckOutcome(cce.NonNull(handler), CommandLineOptions.Clo.ErrorLimit);
-        }
-        catch (UnexpectedProverOutputException e)
-        {
+        } catch (UnexpectedProverOutputException e) {
           outputExn = e;
+        } catch (OperationCanceledException) {
+          outcome = ProverInterface.Outcome.Canceled;
+          // We don't want to log the comment as the process has been killed
+          throw;
```

Comment:
Would the code still work if you remove the `throw`? If not I wouldn't set `outcome`.

Reply:
As the comment says, if we don't throw, it fails because it will call `thmProver.LogComment("canceled");`, which will fail because it will try to send a comment to the Z3 process and the process is not valid anymore since it was killed.

The default value of "outcome" is "valid". If I do not set it to something else, the method `Checker.ReadOutcome` will return `valid`, which will makes its two callers potentially take bad decisions. I don't think these two are in the same thread that is being killed.
Your remark however gave me the indication I also need to update the outcome for the split, as well as the API for the callback.